CRM/Contact/BAO/Query.php

```diff
@@ -680,15 +697,12 @@ public function selectClause($apiEntity = NULL) {
         }
       }
 
-      if (in_array($name, array('prefix_id', 'suffix_id', 'gender_id', 'communication_style_id'))) {
```

Comment:
Consolidate this logic with general option-value pseudoconstant metadata based evaluation further down